### PR TITLE
Added container name in the back-off event log

### LINKED ISSUE #93575: "Back-off restarting failed container": log which container failed
**What would you like to be added**:

The event `Back-off restarting failed container` should become `Back-off restarting failed container <CONTAINER_NAME>`

**Why is this needed**:

Its hard to tell which container failed, you need to dig through the describe output and find something like

```
    State:          Waiting
      Reason:       CrashLoopBackOff
```

### PR DESCRIPTION
**Pull request by senthilnathan**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fixes an issue.

**Which issue(s) this PR fixes**:

Fixes #93575 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -854,7 +854,7 @@ func (m *kubeGenericRuntimeManager) doBackOff(pod *v1.Pod, container *v1.Contain
 	key := getStableKey(pod, container)
 	if backOff.IsInBackOffSince(key, ts) {
 		if ref, err := kubecontainer.GenerateContainerRef(pod, container); err == nil {
-			m.recorder.Eventf(ref, v1.EventTypeWarning, events.BackOffStartContainer, "Back-off restarting failed container")
+			m.recorder.Eventf(ref, v1.EventTypeWarning, events.BackOffStartContainer, "Back-off restarting failed container %s", container.Name)
```
Comment (liggitt) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go:857]: /hold

This increases the back off event writes per pod … the Kubelet throttles event writes and simply omits events over a certain threshold, so increasing the number and variety of events this way could easily drop any events fro other pods

**Comment by SergeyKanzhelev** [comment]
/unassign @dchen1107
/assign @dashpole 
/sig instrumentation

@dashpole what do you think the right fix for this would be? increase log level? Correlate event with the log? I would assume that the original issue reported by @howardjohn from some Istio-related scenario where (I assume) it's important to quickly identify whether it's a sidecar issue or main container's fault.

**Comment by howardjohn** [comment]
> from some Istio-related scenario where (I assume) it's important to quickly identify whether it's a sidecar issue or main container's fault.

That was the use case, yes. Its not impossible to find out if you dig through some of the other output of `kubectl describe` but its certainly an improvement. I thought it would be a lot more trivial to implement though :slightly_smiling_face:

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go (hunk 1 above)
Comment (dashpole) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go:857]: Is this actually the case?  The ref provided looks like it is already at the container granularity, so i'm not certain this actually increases the number or variety of events.  But then again, i'm not familiar with the deduplication logic.

Comment (SergeyKanzhelev) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go:857]: Fast code search results. It looks like the max events count is calculated based on localKey:

https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/client-go/tools/record/events_cache.go#L266-L268

Local key is the message (second return value in here):

https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/client-go/tools/record/events_cache.go#L156-L173

So this change may in fact make everything worse - more events may be lost.

Given that you can get the name of failed container with `describe` command, closing this off for now.

/close

**Pull request closed without merging**